src/cloudant/database.py

```diff
@@ -139,18 +140,36 @@ def delete(self):
 
     def all_docs(self, **kwargs):
         """
-
-        descending  Return the documents in descending by key order boolean false
-        endkey  Stop returning records when the specified key is reached string
-        include_docs    Include the full content of the documents in the return boolean false
-        inclusive_end   Include rows whose key equals the endkey  boolean true
-        key Return only documents that match the specified key  string
-        limit   Limit the number of the returned documents to the specified number  numeric
-        skip    Skip this number of records before starting to return the results  numeric 0
+        TODO: docstring worth a damn
+
+        descending  Return the documents in descending by key
+            order boolean false
+        endkey  Stop returning records when the specified key is
+            reached string
+        include_docs    Include the full content of the documents
+            in the return boolean false
+        inclusive_end   Include rows whose key equals the endkey
+            boolean true
+        key Return only documents that match the
+            specified key  string
+        limit   Limit the number of the returned documents
+            to the specified number  numeric
+        skip    Skip this number of records before starting to
+           return the results  numeric 0
         startkey
 
         """
-        resp = self._r_session.get(posixpath.join(self.database_url, '_all_docs'), params=dict(kwargs))
+        for k in kwargs:
+            if k not in ALL_ARGS:
+                raise ValueError("Invalid argument: {0}".format(k))
```

Comment:
I think that you should fold this loop into python_to_couch. I think that the logic in python_to_couch is potentially a bottleneck, and it would be nice to fold all the looping and potential targets for optimization into that one method, to make it easy to refactor later.